solidity/contracts/bridge/BridgeGovernance.sol

```diff
@@ -0,0 +1,1523 @@
+        bridge.setVaultStatus(vault, isTrusted);
+    }
+
+    /// @notice Begins the governance delay update process.
+    /// @dev Can be called only by the contract owner.
+    /// @param _newGovernanceDelay New governance delay
+    function beginGovernanceDelayUpdate(uint256 _newGovernanceDelay)
+        external
+        onlyOwner
+    {
+        governanceDelays[1] = _newGovernanceDelay;
+        /* solhint-disable not-rely-on-time */
+        governanceDelays[2] = block.timestamp;
+        emit GovernanceDelayUpdateStarted(_newGovernanceDelay, block.timestamp);
+        /* solhint-enable not-rely-on-time */
+    }
+
+    /// @notice Finalizes the governance delay update process.
+    /// @dev Can be called only by the contract owner, after the governance
+    ///      delay elapses.
+    function finalizeGovernanceDelayUpdate() external onlyOwner {
+        require(governanceDelays[2] > 0, "Change not initiated");
+        /* solhint-disable not-rely-on-time */
+        require(
+            block.timestamp - governanceDelays[2] >= governanceDelays[0],
+            "Governance delay has not elapsed"
+        );
+        /* solhint-enable not-rely-on-time */
+        emit GovernanceDelayUpdated(governanceDelays[1]);
+        governanceDelays[0] = governanceDelays[1];
+        governanceDelays[1] = 0;
+        governanceDelays[2] = 0;
+    }
+
+    /// @notice Begins the Bridge governance transfer process.
+    /// @dev Can be called only by the contract owner. It is the governance
+    ///      resposibility to validate the corectness of the new Bridge
+    ///      Governance contract.
+    function beginBridgeGovernanceTransfer(address _newBridgeGovernance)
+        external
+        onlyOwner
+    {
+        newBridgeGovernance = _newBridgeGovernance;
+        /* solhint-disable not-rely-on-time */
+        bridgeGovernanceTransferChangeInitiated = block.timestamp;
+        emit BridgeGovernanceTransferStarted(
+            _newBridgeGovernance,
+            bridgeGovernanceTransferChangeInitiated
+        );
+        /* solhint-enable not-rely-on-time */
+    }
+
+    /// @notice Finalizes the bridge governance transfer process.
+    /// @dev Can be called only by the contract owner, after the governance
+    ///      delay elapses.
+    function finalizeBridgeGovernanceTransfer() external onlyOwner {
+        require(
+            bridgeGovernanceTransferChangeInitiated > 0,
+            "Change not initiated"
+        );
+        /* solhint-disable not-rely-on-time */
+        require(
+            block.timestamp - bridgeGovernanceTransferChangeInitiated >=
+                governanceDelays[0],
+            "Governance delay has not elapsed"
+        );
+        /* solhint-enable not-rely-on-time */
+        bridge.transferGovernance(newBridgeGovernance);
+        emit BridgeGovernanceTransferred(newBridgeGovernance);
+        bridgeGovernanceTransferChangeInitiated = 0;
```

Comment:
We probably want to zero `newBridgeGovernance` as well:
```
newBridgeGovernance = address(0);
```

Reply:
Yea, I went back and forth with it. I agree it is better to assign `address(0)`, but there was a battle around the contract size limit. I don't see an issue if we do not reset it back to `address(0)` other than for informational purposes, and that's why I decided to trim it to go down with the size. However, I see that we have a bit of room for this guy, so here it's added back: [69e4c42](https://github.com/keep-network/tbtc-v2/pull/329/commits/69e4c42b9bdc1c4ef890e6671366f8a5985754d9)